packages/cactus-plugin-keychain-aws-sm/docs/architecture/delete-keychain-entry-endpoint.puml

```diff
@@ -0,0 +1,31 @@
+@startuml Sequence Diagram - Transaction
+
+title Hyperledger Cactus\nSequence Diagram\nDelete Keychain Entry Endpoint
+
+skinparam sequenceArrowThickness 2
+skinparam roundcorner 20
+skinparam maxmessagesize 120
+skinparam sequenceParticipant underline
+
+box "Users" #LightBlue
+actor "User A" as a
+end box
+
+box "Hyperledger Cactus" #LightGray
+entity "API Client" as apic
+entity "API Server" as apis
+end box
+
+box "AWS SM Connector" #LightGreen
+database "AWS SM" as awssm
+end box
+
+a --> apic : Tx DeleteKeychainEntryV1
```

Comment:
@jagpreetsinghsasan Should we call this something else other than `Tx` (implies transaction to me). I'm wondering because if it's a transaction then it would mean that it can be rolled back for example, but I think the keychain implementation right now is closer to "fire and forget" from the perspective of the HTTP request sender because they cannot combine their request with other requests to achieve some sort of "all or nothing" set of operations. They just send their request and then the deletion will either succeed or fail and that's it.
The same question applies to the other 3 diagrams as well. Please let me know your thoughts!

Reply:
I think this should be referred as Tx (transaction) itself.
My point of view on this is that ledger functions ultimately boils down to similar read/write/get/delete sort of operations.
From rollback point, it can be assumed that we aren't using the correct version of these keychains or they themselves dont support transaction rollback.
A very example to support my view is the kv engine 2 for Hashicorp Vault (which do support rollback functionality due to version-control of the keys stored)
https://www.vaultproject.io/docs/secrets/kv
![image](https://user-images.githubusercontent.com/56861721/161243963-6fd0fb2c-3502-499c-ad1c-3e61fbd93c0f.png)

Reply:
@petermetz what is your opinion on this?

Reply:
> @petermetz what is your opinion on this?

@jagpreetsinghsasan Yup, works for me, thanks for expanding on it